task/options/options.go

```diff
@@ -40,22 +45,129 @@ type Options struct {
 
 	// Every represents a fixed period to repeat execution.
 	// this can be unmarshaled from json as a string i.e.: "1d" will unmarshal as 1 day
-	Every time.Duration `json:"every,omitempty"`
+	Every Duration `json:"every,omitempty"`
 
 	// Offset represents a delay before execution.
 	// this can be unmarshaled from json as a string i.e.: "1d" will unmarshal as 1 day
-	Offset *time.Duration `json:"offset,omitempty"`
+	Offset *Duration `json:"offset,omitempty"`
 
 	Concurrency *int64 `json:"concurrency,omitempty"`
 
 	Retry *int64 `json:"retry,omitempty"`
 }
 
+type Duration struct {
+	Node ast.DurationLiteral
+}
+
+func (a Duration) String() string {
+	return ast.Format(&a.Node)
+}
+
+func (a *Duration) Parse(s string) error {
+	q, err := parseSignedDuration(s)
+	if err != nil {
+		return err
+	}
+	a.Node = *q
+	return nil
+}
+
+func MustParseDuration(s string) (dur *Duration) {
+	dur = &Duration{}
+	if err := dur.Parse(s); err != nil {
+		panic(err)
+	}
+	return dur
+}
+
+// parseSignedDuration will convert a string into a possibly negative DurationLiteral.
+// TODO(docmerlin): switch to the one in flux
+func parseSignedDuration(lit string) (*ast.DurationLiteral, error) {
+	r, s := utf8.DecodeRuneInString(lit)
+	if r == '-' {
+		d, err := parser.ParseDuration(lit[s:])
+		if err != nil {
+			return nil, err
+		}
+		for i := range d.Values {
+			d.Values[i].Magnitude = -d.Values[i].Magnitude
+		}
+		d.BaseNode = ast.BaseNode{}
+		return d, nil
+	}
+	d, err := parser.ParseDuration(lit)
+	if err != nil {
+		return nil, err
+	}
+	d.BaseNode = ast.BaseNode{}
+	return d, err
+}
+
+func (a *Duration) UnmarshalText(text []byte) error {
+	q, err := parseSignedDuration(string(text))
+	if err != nil {
+		return err
+	}
+	a.Node = *q
+	return nil
+}
+
+func (a Duration) MarshalText() ([]byte, error) {
+	return []byte(a.String()), nil
+}
+
+func (a Duration) MartialJSON() ([]byte, error) {
+	q := a.String()
+	res := make([]byte, len(q)+2)
+	res[0] = '"'
+	res[len(res)-1] = '"'
+	copy(res[1:len(res)-1], q)
+	return res, nil
```

Comment:
This would probably be simpler as `return json.Marshal(a.String())`...

Since the method name is misspelled, json.Marshal doesn't look at this method, but it does see `MarshalText`:

> If no MarshalJSON method is present but the value implements encoding.TextMarshaler instead, Marshal calls its MarshalText method and encodes the result as a JSON string.

So maybe you can just remove the MarshalJSON and UnmarshalJSON methods altogether?